keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c

```diff
@@ -0,0 +1,254 @@
+    static const char PROGMEM raw_cat_yawn[] = {
+        0x00, 0x38, 0xe4, 0x04, 0xe4, 0xe4, 0xe8, 0xc8, 0x10, 0x20, 0x20, 0x40, 0x40, 0x40, 0x20, 0x20, 
+        0x10, 0x10, 0x10, 0x10, 0x10, 0x18, 0x1c, 0x1a, 0x45, 0xc1, 0x0e, 0xf0, 0x00, 0x00, 0x00, 0x00, 
+        0x00, 0x00, 0x03, 0x0e, 0x11, 0xe7, 0x8f, 0x9f, 0x9f, 0xc0, 0x00, 0x00, 0x00, 0x02, 0x83, 0xe1, 
+        0xf1, 0xf0, 0xf8, 0xfc, 0xfd, 0xfd, 0xfc, 0xfc, 0xfc, 0xfc, 0xf8, 0xf0, 0xe1, 0xc2, 0x0c, 0xf0, 
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x0f, 0xff, 0x39, 0x00, 0x00, 0x00, 0x00, 0x0f, 0x1f, 
+        0x3f, 0x3f, 0x7f, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x7f, 0x7f, 0x3f, 0x0f, 0x83, 0x70, 0x0f, 
+        0x00, 0x00, 0x00, 0x80, 0x60, 0x18, 0x06, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0e, 0x1e, 
+        0xfc, 0xfc, 0xfc, 0xfc, 0xfc, 0xfc, 0xfc, 0xfc, 0xfc, 0xfc, 0x82, 0x7e, 0xc1, 0x00, 0x00, 0x00, 
+        0xc0, 0x38, 0x06, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 
+        0x1f, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xfe, 0xc1, 0xfe, 0x00, 0x00, 
+        0x07, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 
+        0x00, 0x07, 0x07, 0x07, 0x07, 0x07, 0x07, 0x07, 0x07, 0x07, 0x07, 0x07, 0x07, 0x07, 0x00, 0x00
+    };
+
+    oled_write_raw_P(raw_cat_yawn, sizeof(raw_cat_yawn));
+}
+
+
+#if defined(OLED_ENABLE)
+
+/* oled fix for 'cold boot' */
+void keyboard_post_init_user(void) {
+    wait_ms(2000);
+    oled_init(OLED_ROTATION_0);
+}
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    return OLED_ROTATION_90;  
+}
+
+/* change image on keypress tracker */
+bool oled_tracker = false;
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        oled_tracker ^= 1;
+    }
+    return true;
+}
+
+bool oled_task_user(void) {
+
+    switch (get_highest_layer(layer_state)) {
+        case _BASE:
+            oled_set_cursor(0,0);
+            render_base_layer();
+            oled_set_cursor(1,5);
+            oled_write_ln_P(PSTR("Base"), false);
+            break;
+        case _FUNC:
+            oled_set_cursor(0,0);
+            render_func_layer();
+            oled_set_cursor(1,5);
+            oled_write_ln_P(PSTR("Func"), false);
+            break;
+        case _GAME:
+            oled_set_cursor(0,0);
+            render_game_layer();
+            oled_set_cursor(1,5);
+            oled_write_ln_P(PSTR("Game"), false);
+            break;
+        case _CODE:
+            oled_set_cursor(0,0);
+            render_code_layer();
+            oled_set_cursor(1,5);
+            oled_write_ln_P(PSTR("Code"), false);
+            break;
+        default:
+            oled_write_ln_P(PSTR("Undefined"), false);
+    }
+
+    oled_set_cursor(0,11);
+    if (oled_tracker) {
+        render_cat();
+    } else {
+        render_cat_yawn();
+    }
+
+    return false;
+}
+#endif
```

Comment:
This should be moved to the keyboard level in an `ndk16.c` file using the appropriate `_kb` functions, as OLEDs are enabled at the keyboard level.
```suggestion
```

Reply:
Added ndk16.c, moved the OLED code there and changed functions to _kb.

Thanks